mhs/common/mhs_common/workflow/common.py

```diff
@@ -34,3 +40,33 @@ class CommonWorkflow(abc.ABC):
         :param payload: payload to handle
         """
         pass
+
+    @staticmethod
+    def handle_spine_fault(error: httpclient.HTTPClientError,
+                           logger: IntegrationAdaptorsLogger) -> Tuple[int, AnyStr]:
+        """
+
+        :param error: Tornado's HTTPClient error instance
+        :param logger: Logger to be used to log Spine error(s)
+        :return: Response to external client represented as HTTP status code and body
+        """
+        response_code = error.response.code
+        response_headers = error.response.headers
+        response_body = error.response.body
+        client_message = 'Error(s) returned from Spine. Contact system administrator.'
+
+        if response_code == 500:
+            assert 'Content-Type' in response_headers
+            if response_headers['Content-Type'] == 'text/xml':
+                parsed_body = ElementTree.fromstring(response_body)
+                default_fields =  {'errorType': 'ebxml_error'}
+
+                if SOAPFault.is_soap_fault(parsed_body):
+                    fault: SOAPFault = SOAPFault.from_parsed(response_headers, parsed_body)
+
+                    for error in fault.error_list:
+                        all_fields = {**error, **default_fields}
+                        err_text = ', '.join([f'{k}={v}' for k, v in all_fields.items()])
```

Comment:
Optional: The logger actually manages this formatting for you, if you check elsewhere you'll be to see how it's used, it might be neater to use that

Reply:
Changed.